legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py

```diff
@@ -497,9 +498,14 @@ def check_authorization(identifier, filing_json: dict,
 
         # While filing IA business object will be None. Setting default values in that case.
         state = business.state if business else Business.State.ACTIVE
+        if business:
+            legal_type = business.legal_type
+        # for temporary business notice of withdraw, get legalType from filing json
+        elif filing_type == CoreFiling.FilingTypes.NOTICEOFWITHDRAWAL.value:
+            legal_type = filing_json['filing'].get('business', None).get('legalType')
```

Comment:
Does this work if the NoW is for a regular business? (eg, withdrawing a FE alteration)

Reply:
If it's a regular business, the `business` passed into `check_authroization` will not be none, so, we will get `legal_type` from its Business Object. 

And, yes, technically, if it's a regular business, this also works. But, I was thinking about try to keep the original logic as much as possible. In case, there are some special cases for other filings would be better to get the legal_type value from its business table.